plugins/alembic/module/vtkF3DAlembicReader.cxx

```diff
@@ -0,0 +1,148 @@
+      Alembic::AbcGeom::Int32ArraySamplePtr counts = samp.getFaceCounts();
+
+      size_t P_size = positions->size();
+      size_t counts_size = counts->size();
+
+      for (size_t i = 0; i < P_size; i++)
+      {
+        points->InsertNextPoint(
+          positions->get()[i].x, positions->get()[i].y, positions->get()[i].z);
+      }
+
+      size_t face_index = 0;
+      for (size_t i = 0; i < counts_size; i++)
+      {
+        auto polyface_vertex_count = counts->get()[i];
+        polys->InsertNextCell(polyface_vertex_count);
+        for (auto j = 0; j < polyface_vertex_count; j++)
+        {
+          polys->InsertCellPoint(indices->get()[face_index++]);
+        }
+      }
+    }
+    polydata->SetPoints(points);
+    polydata->SetPolys(polys);
+
+    return polydata;
+  }
+
+  void IterateIObject(vtkAppendPolyData* append, const Alembic::Abc::IObject& parent,
+    const Alembic::Abc::ObjectHeader& ohead)
+  {
+    // Set this if we should continue traversing
+    Alembic::Abc::IObject nextParentObject;
+
+    if (Alembic::AbcGeom::IXform::matches(ohead))
+    {
+      Alembic::AbcGeom::IXform xform(parent, ohead.getName());
+      nextParentObject = xform;
+    }
+    else if (Alembic::AbcGeom::IPolyMesh::matches(ohead))
+    {
+      Alembic::AbcGeom::IPolyMesh polymesh(parent, ohead.getName());
+      append->AddInputData(ProcessIPolyMesh(polymesh));
+      nextParentObject = polymesh;
+    }
+
+    // Recursion
+    if (nextParentObject.valid())
+    {
+      for (size_t i = 0; i < nextParentObject.getNumChildren(); ++i)
+      {
+        IterateIObject(append, nextParentObject, nextParentObject.getChildHeader(i));
+      }
+    }
+  }
+
+  void ImportRoot(vtkAppendPolyData* append)
+  {
+    Alembic::Abc::IObject top = this->Archive.getTop();
+
+    for (size_t i = 0; i < top.getNumChildren(); ++i)
+    {
+      IterateIObject(append, top, top.getChildHeader(i));
+    }
+  }
+
+  void ReadScene(const std::string& filePath)
+  {
+    Alembic::AbcCoreFactory::IFactory factory;
+    Alembic::AbcCoreFactory::IFactory::CoreType core_type;
+
+    this->Archive = factory.getArchive(filePath, core_type);
+  }
+  Alembic::Abc::IArchive Archive;
+};
+
+vtkStandardNewMacro(vtkF3DAlembicReader);
+
+//----------------------------------------------------------------------------
+vtkF3DAlembicReader::vtkF3DAlembicReader()
+  : Internals(new vtkF3DAlembicReader::vtkInternals())
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//----------------------------------------------------------------------------
+vtkF3DAlembicReader::~vtkF3DAlembicReader() = default;
+
+//----------------------------------------------------------------------------
+int vtkF3DAlembicReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  this->Internals->ReadScene(this->FileName);
+
+  vtkNew<vtkAppendPolyData> append;
```

Comment:
does that work when working with textures ?

Reply:
yes it will use the same textures for all submeshes.

Reply:
is that ok ? would be beneficial to have @nyue input on this

Reply:
Well, that's the case with all geometry-only formats.  
As far as I know, Alembic supports texture coordinates but no materials. The texture coordinates are supposed to be used in a later stage in Blender/Maya or any final rendering software by crafting materials there.  
The goal of f3d here is to provide a way to visualize these texture coordinates by using a checkboard texture for example.

Reply:
Technically, Alembic does have a [prototype](https://github.com/alembic/alembic/tree/master/lib/Alembic/AbcMaterial) material support but most studio working with Alembic have their own way to deal with what they call *Look Development* in their pipeline and the industry is gravitating towards [MaterialX](https://materialx.org/).

I believe for a start, what @Meakk solution provides is sufficient.

Reply:
Alright then!